src/ts/ConvivaAnalytics.ts

```diff
@@ -186,89 +218,73 @@ export class ConvivaAnalytics {
     return this.sessionKey !== Conviva.Client.NO_SESSION_KEY;
   }
 
-  private onSourceLoaded = (event: any) => {
+  private onSourceLoaded = () => {
     if (this.isAd) {
       // Ignore ON_SOURCE_LOADED events during ad playback, because that's just an ad being temporarily loaded
       // instead of the actual source.
       return;
     }
 
-    if (this.isValidSession()) {
-      // Do not start a new session when a session is already existing
-      // Happens after ad playback, when the actual source is restored and an ON_SOURCE_LOADED event issued. Because
-      // we suppress the ON_SOURCE_UNLOADED event which unloads the temporary ad source, we must also ignore this
-      // event. By ignoring these ad-induced events, we end up with a clean ON_SOURCE_LOADED/ON_SOURCE_UNLOADED
-      // sequence which only concerns the actual source.
-      return;
-    }
+    // Do not start a new session when a session is already existing
+    // Happens after ad playback, when the actual source is restored and an ON_SOURCE_LOADED event issued. Because
+    // we suppress the ON_SOURCE_UNLOADED event which unloads the temporary ad source, we must also ignore this
+    // event. By ignoring these ad-induced events, we end up with a clean ON_SOURCE_LOADED/ON_SOURCE_UNLOADED
+    // sequence which only concerns the actual source.
+    this.updateSession();
```

Comment:
Shouldn't you initialize a new session when a new source is loaded?

Reply:
👍